### PR TITLE
Feature #13057: Make GameAction flags accessible by plugins.

### PR DESCRIPTION
**Pull request by Sadret**

### CODE REVIEW

File: src/openrct2/world/TileElement.h [hunk 1]
```diff
@@ -384,9 +384,9 @@ struct SmallSceneryElement : TileElementBase
 {
 private:
     ObjectEntryIndex entryIndex; // 4
-    uint8_t age;                 // 5
-    uint8_t colour_1;            // 6
-    uint8_t colour_2;            // 7
+    uint8_t age;                 // 6
+    uint8_t colour_1;            // 7
+    uint8_t colour_2;            // 8
```
Comment (AaronVanGeffen) [inline src/openrct2/world/TileElement.h:389]: Why did you increment the address comments here?

Comment (IntelOrca) [reply src/openrct2/world/TileElement.h:389]: ObjectEntryIndex is 2 bytes, so I guess just correcting the comments.

Comment (ZehMatt) [reply src/openrct2/world/TileElement.h:389]: I think so too, they have no real meaning anymore.

Comment (Sadret) [reply src/openrct2/world/TileElement.h:389]: Yes, it was just a small correction of the comments.
I wont recommend to remove them, they are still relevant for the data padding and if you want to access the raw data from within a plugin.

Comment (ZehMatt) [reply src/openrct2/world/TileElement.h:389]: That sounds evil and shouldn't be done, all fields are explicitly exported, please remove the comments.

File: src/openrct2/actions/BannerPlaceAction.hpp [hunk 2]
```diff
@@ -35,6 +35,7 @@ DEFINE_GAME_ACTION(BannerPlaceAction, GAME_COMMAND_PLACE_BANNER, GameActionResul
 
     void AcceptParameters(GameActionParameterVisitor & visitor) override
     {
+        GameAction::AcceptParameters(visitor);
```
[the remark this answers is not in the document]
Comment (Sadret) [reply src/openrct2/actions/BannerPlaceAction.hpp:38]: Also in L1092, or not?

File: src/openrct2/world/TileElement.h (hunk 1 above)
Comment (Sadret) [reply src/openrct2/world/TileElement.h:389]: No, currently there are a lot of fields that are not exported.
I personally have to rely on raw data access at several points in my plugins.

Anyhow, there is still the argument that it makes the padding more transparent.

Comment (Sadret) [reply src/openrct2/world/TileElement.h:389]: But if this is such a big issue maybe I should move it to a different PR.

File: src/openrct2/actions/BannerPlaceAction.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (duncanspumpkin) [reply src/openrct2/actions/BannerPlaceAction.hpp:38]: Yes also in L1092

File: src/openrct2/world/TileElement.h (hunk 1 above)
Comment (duncanspumpkin) [reply src/openrct2/world/TileElement.h:389]: @Sadret just tell us whats missing and we can expose it properly.

Comment (Sadret) [reply src/openrct2/world/TileElement.h:389]: I planned to implement it myself, once I finished the scenery plugin.
At least for the scenery place and remove actions.

File: distribution/changelog.txt [hunk 3]
```diff
@@ -40,6 +40,7 @@
 - Feature: [#12885] Add SmallSceneryElement.quadrant to the plugin API.
 - Feature: [#12886] Make all scenery placement and remove actions available to the plugin API.
 - Feature: [#2350, #12922] Add snow, heavy snow and blizzard to weather types.
+- Feature: [#13057] Make GameAction flags accessible by plugins.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:43]: This is in the wrong place. You added to the 0.3.1 section, which has already been released.

**Review by Gymnasiast — CHANGES_REQUESTED** [review]

**Pull request closed without merging**